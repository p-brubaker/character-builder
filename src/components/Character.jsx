import React from 'react'

function Character() {
  return <div>Character goes here</div>
}

export default Character
