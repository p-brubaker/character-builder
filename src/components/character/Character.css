.character {
  display: grid;
  grid-template-rows: 1.5fr 1fr 2fr;
  height: 700px;
  width: 200px;
}

.character-container {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

.bg {
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
}
