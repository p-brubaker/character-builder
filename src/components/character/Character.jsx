import React from 'react'
import './Character.css'

function Character() {
  return (
    <div className="character-container">
      <div className="character">
        <div className="bg head"></div>
        <div className="bg middle"></div>
        <div className="bg bottom"></div>
      </div>
    </div>
  )
}
export default Character
