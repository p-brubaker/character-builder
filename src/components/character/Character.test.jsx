import { render } from '@testing-library/react'
import Character from './Character'

it('should render the Character component', () => {
  const { container } = render(<Character head="Bird" middle="Fancy" bottom="White" />)
  expect(container).toMatchSnapshot()
})
