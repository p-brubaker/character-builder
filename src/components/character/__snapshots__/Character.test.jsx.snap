// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Character component 1`] = `
<div>
  <div
    class="character-container"
  >
    <div
      class="character"
    >
      <div
        class="bg head"
        style="background-image: url(assets/bird-head.png);"
      />
      <div
        class="bg middle"
        style="background-image: url(assets/fancy-middle.png);"
      />
      <div
        class="bg bottom"
        style="background-image: url(assets/white-pants.png);"
      />
    </div>
  </div>
</div>
`;
